ipywidgets/widgets/widget.py

```diff
@@ -597,10 +623,16 @@ def _should_send_property(self, key, value):
         # A roundtrip conversion through json in the comparison takes care of
         # idiosyncracies of how python data structures map to json, for example
         # tuples get converted to lists.
-        if (key in self._property_lock
-            and jsonloads(jsondumps(to_json(value, self))) == self._property_lock[key]):
-            return False
-        elif self._holding_sync:
+        if key in self._property_lock:
+            # model_state, buffer_paths, buffers
+            split_value = _remove_buffers({ key: to_json(value, self)})
+            split_lock = _remove_buffers({ key: self._property_lock[key]})
+            # Compare state and buffer_paths
+            if (jsonloads(jsondumps(split_value[0])) == jsonloads(jsondumps(split_lock[0]))
```

Comment:
I would rather pass `sort_keys=True` to `dumps` rather than pass it back through `loads`, but it is not critical and I have not profiled the two.